modules/allegroBidAdapter.js

```diff
@@ -0,0 +1,257 @@
+    }
+    if (imp?.secure !== undefined) {
+      imp.secure = imp.secure === 1;
+    }
+    return imp;
+  },
+
+  /**
+   * Builds the full ORTB request and normalizes integer flags. Optionally migrates ext fields
+   * into Google style bracketed keys unless disabled via `allegro.convertExtensionFields` config.
+   *
+   * @param {Function} buildRequest Base builder provided by ortbConverter.
+   * @param {Object[]} imps Array of impression objects.
+   * @param bidderRequest Prebid bidderRequest (contains refererInfo, gdpr, etc.).
+   * @param context Shared converter context.
+   * @returns {Object} Mutated ORTB request object ready to serialize.
+   */
+  request(buildRequest, imps, bidderRequest, context) {
+    const request = buildRequest(imps, bidderRequest, context);
+
+    if (request?.device?.dnt !== undefined) {
+      request.device.dnt = request.device.dnt === 1;
+    }
+
+    if (request?.device?.sua?.mobile !== undefined) {
+      request.device.sua.mobile = request.device.sua.mobile === 1;
+    }
+
+    if (request?.test !== undefined) {
+      request.test = request.test === 1;
+    }
+
+    // by default, we convert extension fields unless the config explicitly disables it
+    const convertExtConfig = config.getConfig('allegro.convertExtensionFields');
+    if (convertExtConfig === undefined || convertExtConfig === true) {
+      convertExtensionFields(request);
+    }
+
+    if (request?.source?.schain && !isSchainValid(request.source.schain)) {
+      delete request.source.schain;
+    }
+
+    return request;
+  }
+})
+
+/**
+ * Validates supply chain object structure
+ * @param schain - Supply chain object
+ * @return {boolean} True if valid, false otherwise
+ */
+function isSchainValid(schain) {
+  try {
+    if (!schain || !schain.nodes || !Array.isArray(schain.nodes)) {
+      return false;
+    }
+    const requiredFields = ['asi', 'sid', 'hp'];
+    return schain.nodes.every(node =>
+      requiredFields.every(field => node.hasOwnProperty(field))
+    );
+  } catch (error) {
+    logError('Allegro: Error validating schain:', error);
+    return false;
+  }
+}
+
+/**
+ * Allegro Bid Adapter specification object consumed by Prebid core.
+ */
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER, VIDEO, NATIVE],
+  gvlid: GVLID,
+
+  /**
+   * Validates an incoming bid object.
+   *
+   * @param bid Prebid bid request params.
+   * @returns {boolean} True if bid is considered valid.
+   */
+  isBidRequestValid: function (bid) {
+    return !!(bid);
+  },
+
+  /**
+   * Generates the network request payload for the adapter.
+   *
+   * @param bidRequests List of valid bid requests.
+   * @param bidderRequest Aggregated bidder request data (gdpr, usp, refererInfo, etc.).
+   * @returns Request details for Prebid to send.
+   */
+  buildRequests: function (bidRequests, bidderRequest) {
+    const url = config.getConfig('allegro.bidderUrl') || BIDDER_URL;
```

Comment:
What's the intent behind the configurable endpoint? is it also the reason why you have `convertExtensionFields`? (it does not look like something a publisher would care about).

We have a rule against "fully variable endpoints", which is to avoid burdening publishers with complicated logic to calculate them at run time (e.g. routing users to a regional endpoint).